### ISSUE TITLE
Basic access control / authorization requirement

### ISSUE THREAD

**Issue by Sjord** [OWASP/ASVS#877]
I think the ASVS currently lacks a requirement that states that you should perform access control or authorisation checks at all. I would expect requirement 4.1.1 to be something like:

> Verify that functionality is only accessible by users that are authorised to use that functionality.

This is implied in many of the requirements in chapter 4, but I think it would be valuable to make this its own requirement. If an application is missing an authorisation check, I want to be able to point to an ASVS requirement that specifies that every function should have access control, without necessarily including least privilege principle or trusted service layers.

**Comment by jmanico** [comment]
Hey there. 4.1.3 captures the POLP principle well.

4.1.3 Verify that the principle of least privilege exists - users should only be able to access functions, data files, URLs, controllers, services, and other resources, for which they possess specific authorization. This implies protection against spoofing and elevation of privilege. 

--
Jim Manico
@Manicode
Secure Coding Education
+1 (808) 652-3805

**Comment by elarlang** [comment]
By @Sjord 's description it's more architecture requirement but similar requirement exists there also:

> V1.4.3 Verify enforcement of the principle of least privilege in functions, data files, URLs, controllers, services, and other resources. This implies protection against spoofing and elevation of privilege.

**Comment by jmanico** [comment]
1.4.3 and 4.1.3 seems like duplicates. There’s only so many ways you can talk about the POLP principal, it’s not a complex idea.

--
Jim Manico

**Comment by jmanico** [comment]
May I suggest we delete 1.4.3 or 4.1.3? @elarlang ? @Sjord ?

**Comment by elarlang** [comment]
From the original issue point of view - I don't think we need another requirement for "verify that permission checks are there".
For me there are many duplicates in V4 category, but maybe I have not analyzed them enough yet.

Before deleting 1.4.3 or 4.1.3 (but I think it should be separate issue) it would be nice to know, why one of them is in architecture category and another in authorization category - maybe there is some good reason and instead of deleting it requires improvement in wording.

**Comment by jmanico** [comment]
I think architecture is redundant, I can’t see why we need both.

**Comment by elarlang** [comment]
Rethinking about it - agree with deleting current 1.4.3 as it's duplicate to 4.1.3. It's practically testable, should be level 1, and that's why we need to keep 4.1.3 instead of 1.4.3. If there is no good explanation, why there are some duplicates in architecture categories...

**Comment by jmanico** [comment]
I have no idea why there are so many architect level duplicates and I endorse getting rid of duplicates!

**Comment by lfservin** [comment]
It seems like many architectural requirements for access control (1.4.x) are repeated in the access control chapter (4.1.x). I'll create a separate ticket to list them all

**Comment by jmanico** [comment]
I agree. I suggest we remove them from the architectural section and move them to the access control section and remove redundancy.

Thank you! 🙏

**Comment by elarlang** [comment]
@lfservin - I made "ping" for you in Slack in related thread. The question is - maybe requirements need to be changed to "documentation" requirements and make them clearly separate from "implementation" requirements instead of just removing them.

**Comment by lfservin** [comment]
I'll look there and answer there

**Comment by elarlang** [comment]
Status: waits "scope definition" for V1 category (requirements).
